src/main/java/uk/gov/hmcts/dts/mytime/services/LeaveRequestService.java

```diff
@@ -0,0 +1,63 @@
+package uk.gov.hmcts.dts.mytime.services;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import uk.gov.hmcts.dts.mytime.exceptions.DuplicatedItemException;
+import uk.gov.hmcts.dts.mytime.exceptions.NotFoundException;
+import uk.gov.hmcts.dts.mytime.models.LeaveRequest;
+import uk.gov.hmcts.dts.mytime.models.LeaveStatus;
+import uk.gov.hmcts.dts.mytime.repository.LeaveRequestRepository;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+public class LeaveRequestService {
+    @Autowired
+    private LeaveRequestRepository leaveRequestRepository;
+
+    @Transactional
+    public LeaveRequest createLeaveRequest(LeaveRequest leaveRequest) {
+        uk.gov.hmcts.dts.mytime.entities.LeaveRequest newLeaveRequestEntity =
+            new uk.gov.hmcts.dts.mytime.entities.LeaveRequest(leaveRequest.getEmployeeId(),
+                                                              leaveRequest.getApproverId(),
+                                                              leaveRequest.getType(),
+                                                              LeaveStatus.AWAITING,
+                                                              leaveRequest.getStartDate(),
+                                                              leaveRequest.getEndDate(),
+                                                              leaveRequest.getRequestComment(),
+                                                              leaveRequest.getApproverComment());
+
+        if (leaveRequestRepository.findAllByEmployeeId(leaveRequest.getEmployeeId()).stream()
```

Comment:
Could this check be done first before the entity is created? 

Prob a small thing but then if this gets raised, it means we save 0.000001 seconds on the entity being created when it isn't going to be used 

100000 cases of that, and we would have saved 1 second